mining.md

```diff
@@ -386,25 +264,29 @@ To create a block, first compute a few fields:
 - `MsgRoot` - To compute this:
   - Select a set of messages from the mempool to include in your block.
   - Insert them into a Merkle Tree and take its root.
+- `StateRoot` - Apply each of your chosen messages to the `ParentState` to get this.
 - `ReceiptsRoot` - To compute this:
   - Apply the set of messages selected above to the parent state, collecting invocation receipts as you go.
   - Insert them into a Merkle Tree and take its root.
-- `StateRoot` - Apply each of your chosen messages to the `ParentState` to get this.
 - `BlockSig` - A signature with your private key (must also match the ticket signature) over the entire block. This is to ensure that nobody tampers with the block after we propogate it to the network, since unlike normal PoW blockchains, a winning ticket is found independently of block generation.
 
-Start by filling out `Parents`, `Tickets` and `ElectionProofs` with values from the ticket checking process.
+Start by filling out `Parents`, `Tickets` and `ElectionProof` with values from the ticket checking process.
+
+Next, compute the aggregate state of your selected parent blocks, the `ParentState`. This is done by taking the aggregate parent state of *their* parent tipset, sorting your parent blocks by their tickets, and applying each message in each block to that state. Any message whose nonce is already used (duplicate message) in an earlier block should be skipped (application of this message should fail anyway). Note that re-applied messages may result in different receipts than they produced in their original blocks, an open question is how to represent the receipt trie of this tipsets 'virtual block'. For more details on message execution and state transitions, see the [Filecoin state machine](state-machine.md) document.
+
+Once you have the aggregate `ParentState`, you must apply the mining reward. This is done by adding the correct amount to the miner owners account balance in the state tree. (TODO: link to block reward calculation. Currently, the block reward is a fixed 1000 filecoin).
```

Comment:
typo: miner owner's.

It'd be nice if htere was some special "coinbase" type thing here as opposed to just add to the balance to mark a special event. Just a thought.

Reply:
Also a way to gracefully correct miscalculation here if it happens.